runtime/pack/dist/opt/comment/autoload/comment.vim

```diff
@@ -76,3 +76,84 @@ export def Toggle(...args: list<string>): string
     noautocmd keepjumps setline(lnum1, lines)
     return ''
 enddef
+
+
+# Comment text object
+# Usage:
+#     import autoload 'dist/comment.vim'
+#     onoremap <silent>ic <scriptcmd>comment.ObjComment(v:true)<CR>
+#     onoremap <silent>ac <scriptcmd>comment.ObjComment(v:false)<CR>
+#     xnoremap <silent>ic <esc><scriptcmd>comment.ObjComment(v:true)<CR>
+#     xnoremap <silent>ac <esc><scriptcmd>comment.ObjComment(v:false)<CR>
+export def ObjComment(inner: bool)
+    def IsComment(): bool
+        var stx = map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")')->join()
+        if stx =~ 'Comment'
+            return true
+        else
+            return false
+        endif
+    enddef
+
+    var pos_init = getcurpos()
+
+    # If not in comment, search next one,
+    if !IsComment()
+        if search('\S\+', 'W', line(".") + 100, 100, () => !IsComment()) <= 0
+            return
+        endif
+    endif
+
+    # Search for the beginning of the comment block
+    if IsComment()
+        if search('\v(\S+)|$', 'bW', 0, 200, IsComment) > 0
+            search('\v(\S+)|$', 'W', 0, 200, () => !IsComment())
+        else
+            cursor(1, 1)
```

Comment:
When would this else trigger? I am wondering when we would need to start jumping to the beginning of the file and start searching again.

Reply:
I will add a test case for this

https://asciinema.org/a/WTLvMeBAM1R6Ld7106iTFa0tb